### ISSUE TITLE
Add price_id support for fees API

### ISSUE THREAD

**Issue by mintplugins** [easydigitaldownloads/easy-digital-downloads#4477]
Currently, fees are assigned to the download id but not the variable price. This means that fees get assigned multiple times for items in the cart with variable prices and fees:

For example, say you have a shirt with variable pricing (small medium large) and each gets added to the cart. In the fees for EACH item, all 3 shipping fees are listed under fees for EACH one when getting cart details from **edd_get_payment_meta_cart_details()**

Ideally, we'd be able to do something like this:

```
EDD()->fees->add_fee( array(
    'amount'      => $amount,
    'label'       => sprintf( __( '%s Shipping', 'edd-simple-shipping' ), get_the_title( $item['id'] ) ),
    'id'          => 'simple_shipping_' . $key,
    'download_id' => $item['id'],
    'price_id'    => $item['id']
) );
```

Screenshot of output from edd_get_payment_meta_cart_details()

![screen-shot-2016-04-06-at-4 25 46-pm](https://cloud.githubusercontent.com/assets/1638128/14332601/49934d34-fc18-11e5-9991-feb092c78680.jpg)

**Comment by pippinsplugins** [comment]
This should be a pretty easy fix. Let's make it happen for 2.6.

**Comment by mintplugins** [comment]
This branch needs testing:
https://github.com/easydigitaldownloads/easy-digital-downloads/tree/issue/4477

You could use simple shipping on line 563 by changing the add-fee call to this

`EDD()->fees->add_fee( array(
	'amount'      => $amount,
	'label'       => sprintf( __( '%s Shipping', 'edd-simple-shipping' ), get_the_title( $item['id'] ) ),
	'id'          => 'simple_shipping_' . $key,
	'download_id' => $item['id'],
	'price_id'    => $item['options']['price_id']
) );`

1. Make a "small", "medium", "large" t-shirt.
2. Assign shipping fees for each variable price.
3. Add all 3 sizes to your cart and checkout.

We need to check what gets saved into the database for "_edd_payment_meta" in the post_meta table for this payment. Under "cart_details" > "product" > "fees" it should ONLY have the fee assigned to the variable price rather than ALL fees for all variable prices. Under "Fee"s at the root level of the saved array, it should still have all 3 fees listed.

**Comment by mintplugins** [comment]
Pull Request: https://github.com/easydigitaldownloads/easy-digital-downloads/pull/4494

**Comment by pippinsplugins** [comment]
@mintplugins I'm going to try and test this soon.

**Comment by pippinsplugins** [comment]
Let's make sure we get new unit tests in for testing price_id.

**Comment by pippinsplugins** [comment]
This is working well for me.

@cklosowski could you please give it a once over?

**Comment by cklosowski** [comment]
@pippinsplugins yep, @mintplugins and I are working on some unit test stuff and then i'll verify it again as well.

**Comment by mintplugins** [comment]
@pippinsplugins @cklosowski all tests are passing for me now. Note that I rewrote almost all of the fee tests to be:

- totally fresh for each test
- a little bit more specific 
- use new fee system instead of legacy wherever possible

It's probably a good idea to review those since these are my first tests ever.

**Comment by pippinsplugins** [comment]
Tests look good to me!

**Comment by pippinsplugins** [comment]
Merged into `release/2.6`!
